### PR TITLE
fix #6531 added storage settings into system information

### LINKED ISSUE #6531: Add storage settings to system information
The system information should show the used storage directories:
- db directory 
- geocache data directory 
- public cgeo user storage 

Maybe including space left.

### PR DESCRIPTION
**Pull request by pstorch**

### CODE REVIEW

**Comment by pstorch** [comment]
Looks like this on my device:

```
--- System information ---
Device: ONEPLUS A3003 (OnePlus3, OnePlus)
Android version: 7.1.1
Android build: ONEPLUS A3003_16_170411
C:geo version: 2017.04.29-379acdb developer build
Google Play services: enabled - 10.2.98 (448-146496160)
Low power mode: inactive
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide own/found: false
Map strategy: auto
HW acceleration: enabled (default state)
System language: en_GB
System date format: dd/MM/y
Debug mode active: no
System internal c:geo dir: /data/user/0/cgeo.geocaching (20.0 GB free)
User storage c:geo dir: /storage/emulated/0/cgeo (19.9 GB free)
Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (19.9 GB free)
Database: /storage/emulated/0/Android/data/cgeo.geocaching/files/databases/data (53.7 MB) on user storage
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / PREMIUM
   opencaching.de: Logged in (Login OK)
Geocaching.com date format: dd.MM.yyyy
Installed c:geo plugins: contacts
--- End of system information ---
```

**Comment by Lineflyer** [comment]
I think it should also show if the listed geocache data dir is the internal or external.

**Comment by pstorch** [comment]
@Lineflyer if it is a removable storage it will show it as "removable" at the end (I only have internal/emulated storages).

**Comment by pstorch** [comment]
I made the postfix more specific:
- internal
- external
- external removable
- external non-removable

Only `external` for API level < 21, where we don't have more information.

File: main/src/cgeo/geocaching/storage/DataStore.java [hunk 1]
```diff
@@ -477,7 +477,7 @@ private static File databasePath(final boolean internal) {
     }
 
     @NonNull
-    private static File databasePath() {
+    public static File databasePath() {
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/storage/DataStore.java]: You could leave the methods private by moving the code of appendDirectory and appendDatabase into LocalStorage and DataStore (and probably rename them to something like LocalStorage.appendDirectorySystemInformation(...), appendDatabaseSystemInformation).

However, I'm aware that both alternatives have drawbacks (one needing public access, the other smearing functionality of the system information over multiple classes), so I can't really say what to prefer due to what reasoning.

**Comment by pstorch** [comment]
I thought about it. In the end I decided to just make it public. The other directories are public, too. And we don't expose a secret or a mutable object here. On the other hand I thought about move some logic from DataStore to LocalStorage.

File: main/src/cgeo/geocaching/utils/SystemInformation.java [hunk 2]
```diff
@@ -49,7 +53,7 @@ public static String getSystemInformation(@NonNull final Context context) {
                 .append("\nDevice: ").append(Build.MODEL).append(" (").append(Build.PRODUCT).append(", ").append(Build.BRAND).append(')')
                 .append("\nAndroid version: ").append(VERSION.RELEASE)
                 .append("\nAndroid build: ").append(Build.DISPLAY)
-                .append("\nCgeo version: ").append(Version.getVersionName(context));
+                .append("\nC:geo version: ").append(Version.getVersionName(context));
```
Comment (UniQP) [inline main/src/cgeo/geocaching/utils/SystemInformation.java]: It's `c:geo` also at the beginning of a sentence.

Comment (pstorch) [inline main/src/cgeo/geocaching/utils/SystemInformation.java]: Right, changed it

**Pull request merged**